doc/hooks/versionCheckHook.section.md

```diff
@@ -16,20 +22,67 @@ stdenv.mkDerivation (finalAttrs: {
+
+Below we tabulate the differences between `versionCheckHook` and `testers.testVersion`.
+
+
+| Item                                   | `versionCheckHook`                                                                                                                                            | `testers.testVersion`                                                                                                                                                                                       |
+|:---------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------:|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|
+| Customization                          | Besides the attributes described above, the hook provides no other methods for controlling it.                                                                | The tester has access to all feature set of Nix and Nixpkgs, like generating multiple version test derivations from a list of strings, and accessing programs / tools outside the inputs of the derivation. |
+| Execution environment                  | The hook uses `env --ignore-environment --chdir=/` to run the executables in an environment as clean as possible, but there is no way to change its behavior. | The tester executes in an identical environment of a consumer, independent from the building environment.                                                                                                   |
+| Rebuild after modification             | The hook rebuilds the package when the hook is modified, since it is a phase running during the build time of the package.                                    | The tester does not require rebuilding the package, since it is a derivation.                                                                                                                               |
+| Overhead during package building       | Negligible. Although running during the build time of the package, the hook is lean and executes few commands.                                                | Zero, since the tester is a derivation that runs after package building.                                                                                                                                    |
+| Content-addressed-derivation awareness | Since it runs during the building, the hook does not deal with failures happening after building, like rewritings that happen post installation.              | Since it runs after the building, the tester detects failures at this time.                                                                                                                                 |
```

Comment:
I must say that I don't understand this line of comparison at all. But maybe it is just me, who is not very well familiar with content addressed derivations...

Reply:
The citation of CA will change to something more general.